constance/test/pytest.py

```diff
@@ -0,0 +1,79 @@
+"""
+Pytest constance override config plugin.
+
+Inspired by https://github.com/pytest-dev/pytest-django/.
+"""
+import pytest
+
+from constance import config as constance_config
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_configure(config):  # pragma: no cover
+    """
+    Register override_config marker.
+    """
+    config.addinivalue_line(
+        "markers",
+        (
+            "override_config(**kwargs): "
+            "mark test to override django-constance config"
+        )
+    )
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_call(item):  # pragma: no cover
+    """
+    Validate constance override marker params. Run test with overrided config.
+    """
+    marker = item.get_closest_marker("override_config")
+    if marker is not None:
+        if marker.args:
+            pytest.fail(
+                "Constance override can not not accept positional args"
+            )
+        with ConstanceConfigWrapper(**marker.kwargs):
+            yield
+    else:
+        yield
+
+
+class ConstanceConfigWrapper:
```

Comment:
You could use `ContextDecorator` which would allow to use it as decorator and context manager:
```suggestion
class override_config(ContextDecorator):
```
https://docs.python.org/3/library/contextlib.html#contextlib.ContextDecorator